WalletWasabi.Tests/UnitTests/WabiSabi/Client/CoinJoinCoinSelectionTests.cs

```diff
@@ -0,0 +1,139 @@
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Crypto.Randomness;
+using WalletWasabi.Tests.Helpers;
+using WalletWasabi.WabiSabi.Client;
+using WalletWasabi.WabiSabi.Models;
+using WalletWasabi.WabiSabi.Models.MultipartyTransaction;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.WabiSabi.Client;
+
+public class CoinJoinCoinSelectionTests
+{
+
+	[Fact]
+	public void SelectEmptySetOfCoins()
+	{
+		using SecureRandom rnd = new();
+		var coins = CoinJoinClient.SelectCoinsForRound(
+				coins: Enumerable.Empty<SmartCoin>(),
+				CreateMultipartyTransactionParameters(),
+				consolidationMode: false,
+				minAnonScoreTarget: 10,
+				rnd);
+
+		Assert.Empty(coins);
+	}
+
+	[Fact]
+	public void FullyPrivateSetOfCoins()
+	{
+		// This test is to make sure no coins are selected when all coins are private.
+		const int MinAnonimitySet = 10;
+		var km = KeyManager.CreateNew(out _, "", Network.Main);
+		var hdpubkey = BitcoinFactory.CreateHdPubKey(km);
+		var privateCoins = Enumerable
+			.Range(0, 10)
+			.Select(i => BitcoinFactory.CreateSmartCoin(hdpubkey, Money.Coins(1m), 0, anonymitySet: MinAnonimitySet + 1))
+			.ToList();
+
+		using SecureRandom rnd = new();
+		var coins = CoinJoinClient.SelectCoinsForRound(
+				coins: privateCoins,
+				CreateMultipartyTransactionParameters(),
+				consolidationMode: false,
+				minAnonScoreTarget: MinAnonimitySet,
+				rnd);
+
+		Assert.Empty(coins);
+	}
+
+	[Fact]
+	public void OnlyOneNonPrivateCoinInBigSetOfCoins()
```

Comment:
This test isn't good. It isn't expected behavior.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/7133/commits/1ec012021a11203c495fd3308bf44d5a4f1e8e8f